games/xworld3d/maps/xworld3d_env.py

```diff
@@ -311,49 +314,125 @@ def _configure(self):
         """
         raise NotImplementedError()
 
+    def set_property(self, entity, property_value_dict={}):
+        """
+        Reinstantiate the specified properties of an existing entity.
+        Properties and corresponding values are specified by the property_value_dict in the form
+        of {property : value, ...}, e.g. {"name" : "apple", "loc" : (0, 0)}. value could be None
+        (force reinstantiation) or a valid value for that property.
+        1) If None value is provided for a specified property (e.g. {"name" : None}), entity
+           property will be reinstantiated regardless of its original value.
+        2) otherwise, the value will be assigned to that property of the entity.
+        3) all unset entity properties will be instantiated.
+        """
+        default_dict = OrderedDict.fromkeys(["name", \
+                                             "id", \
+                                             "loc", \
+                                             "asset_path", \
+                                             "yaw"])
+        pv_dict = default_dict.copy()
+        pv_dict.update(property_value_dict)
+
+        # pre-processing for name and asset_path due to their dependency
+        path_value = property_value_dict.get("asset_path", "empty")
+        name_value = property_value_dict.get("name", "empty")
+
+        # if name is specified by user but asset_path is unspecified
+        if name_value != "empty" and path_value == "empty":
```

Comment:
We want to use "empty" to differentiate the cases between properties that are specified and unspecified by the user, e.g.:

1) property_value_dictionary={"name" : None}
2) property_value_dictionary={}

In case 1) , "name" has a None value, indicating the user wants to instantiate its value
In case 2),  "name" is not specified, indicating it remains unchanged unless the entity.name is None